src/controllers/productController.js

```diff
@@ -255,6 +230,107 @@ class productController{
     }
   }
     
+
+  static async availableProducts(req,res){
+    try {
+      if (req.user.roleName !== "vendor") {
+        return res.status(401).json({ status: "fail", message: req.t("Unauthorized") });
+      }
+     
+      const product = await Product.findAll({ where: {vendorId: req.user.id } });
+     
+      if (!product) {
+        return res.status(404).json({ status: "fail", message: req.t("productnotfound")});
+      }
+      return res.status(200).json({ status: req.t("success"),products:{product} });
```

Comment:
- message and data attributes needs to added in the response 
- curl braces around product are not necessarly
- ` return res.status(200).json({ status: req.t("success"),message: "all products", data:product });`